BEIMA.Backend.Test/DeviceFunctions/AddDeviceTest.cs

```diff
@@ -24,13 +30,29 @@ public async Task NoDevice_AddDevice_ReturnsValidId()
                   .Verifiable();
             MongoDefinition.MongoInstance = mockDb.Object;
 
+            Mock<IStorageProvider> mockStorage = new Mock<IStorageProvider>();
+            mockStorage.Setup(mock => mock.PutFile(It.IsAny<IFormFile>()))
+                .Returns(Task.FromResult(Guid.NewGuid().ToString() + ".txt"))
+                .Verifiable();
+
+            StorageDefinition.StorageInstance = mockStorage.Object;
+
             // Create request
-            var body = TestData._testDevice;
-            var request = CreateHttpRequest(RequestMethod.POST, body: body);
-            var logger = (new LoggerFactory()).CreateLogger("Testing");
+            var data = TestData._testDevice;
+            var fileCollection = new FormFileCollection();
+
+            string? deviceId;
+            using (var fileStream = new ByteArrayContent(Encoding.ASCII.GetBytes("TestOne")).ReadAsStream())
```

Comment:
We should add test file content into one of our test data classes.

Reply:
Do you mean just moving the Encoding.ASCII... portion?